packages/report-server/src/reportBuilder/transform/functions/aggregate/aggregations.ts

```diff
@@ -3,75 +3,124 @@
+const sum = (values: FieldValue[]): number | undefined => {
+  if (values.length !== 0) {
+    const checkedValues: number[] = checkIsNum(values);
+    return checkedValues.reduce((a, b) => {
+      return a + b;
+    });
+  }
+  return undefined;
 };
 
-const max = (existingRow: Row, field: string, value: FieldValue) => {
-  const existingValue: FieldValue = existingRow[field];
-  if (!isUndefined(value)) {
-    if (isUndefined(existingValue)) {
-      existingRow[field] = value;
-    } else if (value > existingValue) {
-      existingRow[field] = value;
-    }
+const avg = (values: FieldValue[]): number | undefined => {
+  const numerator: number | undefined = sum(values);
+  const denominator: number = count(values);
+  if (!isUndefined(numerator) && !isUndefined(denominator) && denominator !== 0) {
+    return numerator / denominator;
   }
+  return undefined;
 };
 
-const min = (existingRow: Row, field: string, value: FieldValue) => {
-  const existingValue: FieldValue = existingRow[field];
-  if (!isUndefined(value)) {
-    if (isUndefined(existingValue)) {
-      existingRow[field] = value;
-    } else if (value < existingValue) {
-      existingRow[field] = value;
-    }
+const count = (values: FieldValue[]): number => {
+  return values.length;
+};
+
+const max = (values: FieldValue[]): FieldValue => {
+  if (values.length !== 0) {
+    let maxValue: FieldValue = values[0];
+    values.forEach(value => {
+      if (!isUndefined(value) && !isUndefined(maxValue) && value > maxValue) maxValue = value;
+    });
+    return maxValue;
+  }
+  return undefined;
+};
+
+const min = (values: FieldValue[]): FieldValue => {
+  if (values.length !== 0) {
+    let minValue: FieldValue = values[0];
+    values.forEach(value => {
+      if (!isUndefined(value) && !isUndefined(minValue) && value < minValue) minValue = value;
+    });
+    return minValue;
   }
+  return undefined;
 };
 
-const unique = (existingRow: Row, field: string, value: FieldValue) => {
-  if (!isUndefined(existingRow[field]) && existingRow[field] !== value) {
-    existingRow[field] = 'NO_UNIQUE_VALUE';
-  } else {
-    existingRow[field] = value;
+const unique = (values: FieldValue[]): FieldValue => {
+  if (values.length !== 0) {
+    for (let i = 0; i < values.length - 1; i++) {
+      if (values[i] !== values[i + 1]) {
+        return 'NO_UNIQUE_VALUE';
+      }
+    }
+    return values[0];
   }
+  return undefined;
 };
 
-const drop = (existingRow: Row, field: string, value: FieldValue) => {
+const drop = (values: FieldValue[]): FieldValue => {
   // Do nothing, don't add the field to the existing row
 };
 
-const first = (existingRow: Row, field: string, value: FieldValue) => {
-  if (isUndefined(existingRow[field])) {
-    existingRow[field] = value;
+const first = (values: FieldValue[]): FieldValue => {
+  for (let i = 0; i < values.length; i++) {
+    if (!isUndefined(values[i])) {
+      return values[i];
+    }
   }
+  return undefined;
```

Comment:
```suggestion
  return values.find(isNotNull);
```
You may have noticed that I _hate_ old-fashioned for loops 😝